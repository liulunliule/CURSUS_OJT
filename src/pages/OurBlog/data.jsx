import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchBlogs = () => {
  const [blogs, setBlogs] = useState([]);

  useEffect(() => {
    axios.get("https://668f4a0880b313ba09178dee.mockapi.io/api/blogs")
      .then((response) => setBlogs(response.data))
      .catch((error) => console.error("Error fetching blog data:", error));
  }, []);

  return blogs;
};