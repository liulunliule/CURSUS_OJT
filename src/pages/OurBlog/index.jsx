import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import "./index.scss";
import { blogImage1, blogImage2, blogImage3, blogImage4 } from "../../assets";

const OurBlog = () => {
  const [selected, setSelected] = useState("Blog");
  const [blogs, setBlogs] = useState([]);

  useEffect(() => {
    fetch("https://668f4a0880b313ba09178dee.mockapi.io/api/blogs")
      .then((response) => response.json())
      .then((data) => {
        const updatedData = data.map(blog => {
          switch(blog.Blog_ID) {
            case 1:
              return { ...blog, Blog_Image: blogImage1 };
            case 2:
              return { ...blog, Blog_Image: blogImage2 };
            case 3:
              return { ...blog, Blog_Image: blogImage3 };
            case 4:
              return { ...blog, Blog_Image: blogImage4 };
            default:
              return blog;
          }
        });
        setBlogs(updatedData);
      })
      .catch((error) => console.error("Error fetching blog data:", error));
  }, []);

  const handleSelect = (item) => {
    if (item === "About") {
      window.location.href = "/secondLayout/about";
    } else if (item === "Company") {
      window.location.href = "/secondLayout/company";
    } else if (item === "Blog") {
      window.location.href = "/secondLayout/blog";
    } else if (item === "Careers") {
      window.location.href = "/secondLayout/careers";
    } else if (item === "Press") {
      window.location.href = "/secondLayout/press";
    }
    setSelected(item);
  };

  return (
    <div className="blog-page">
      <header className="custom-header" style={{ backgroundColor: "#f8f8f8" }}>
        <nav className="custom-nav">
          <ul className="custom-ul">
            {["About", "Blog", "Company", "Careers", "Press"].map((item) => (
              <li
                key={item}
                className={`custom-li ${selected === item ? "selected" : ""}`}
                onClick={() => handleSelect(item)}
              >
                {item}
              </li>
            ))}
          </ul>
        </nav>
      </header>

      <section className="hero" style={{ backgroundColor: "#f8f8f8" }}>
        <h1>Insights, ideas, and stories</h1>
      </section>

      <section className="content">
        <div className="content-main">
          <div className="sidebar">
            <div className="left">
              <div className="search">
                <input type="text" placeholder="Search" />
              </div>
              <div className="labels">
                <h3>Labels</h3>
              </div>
              <div className="archive">
                <h3>Archive</h3>
              </div>
              <div className="follow-buttons">
                <button className="follow1">Follow our Facebook page</button>
                <button className="follow2">Follow our X page</button>
              </div>
              <a href="#" className="learn-more">
                Learn More
              </a>
              <a href="#" className="help-center">
                Cursus Help Center
              </a>
            </div>
          </div>

          <div className="blog-list">
            {blogs.map((blog) => (
              <div className="blog-item" key={blog.Blog_ID}>
                <img src={blog.Blog_Image} alt={`Blog ${blog.Blog_ID}`} />
                <div className="blog-info">
                  <p>{blog.Views} â€¢ {blog.Create_At}</p>
                  <h3>{blog.Blog_Title}</h3>
                  <p>{blog.Intro}</p>
                  <Link to="/secondLayout/blog-detail">Read More</Link>
                </div>
              </div>
            ))}
          </div>
        </div>
      </section>
    </div>
  );
};

export default OurBlog;