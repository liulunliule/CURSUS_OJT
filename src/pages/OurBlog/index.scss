.blog-page {
  font-family: Arial, sans-serif;
  margin-top: 56px;
  color: #333;

  p {
    color: gray;
    margin-bottom: 30px;
  }
  
  .custom-header {
    background: #f8f8f8;
    padding: 10px 0;
  }
  
  .custom-nav {
    .custom-ul {
      list-style: none;
      display: flex;
      justify-content: center;
      gap: 20px;
      padding: 0;
      margin: 0;
  
      .custom-li {
        cursor: pointer;
        padding: 10px;
        position: relative;
  
        &.selected::after {
          content: '';
          position: absolute;
          bottom: -3px;
          left: 0;
          right: 0;
          height: 3px;
          background: red;
        }
  
        &:hover {
          font-weight: bold;
        }
      }
    }
  }

  .content {
    padding: 20px;

    h1 {
      text-align: center;
      margin-bottom: 20px;
    }

    .content-main {
      display: flex;

      .sidebar {
        width: 30%;
        padding-right: 20px;

        .left{
          background-color: #fff;
          padding: 10px 0;
        }

        .search {
          display: flex;
          flex-direction: column;
          margin-left: 106px;
          margin-bottom: 20px;
          width: 50%;

          input {
            width: 100%;
            padding: 10px;
            height: 40px;
            border: 1px solid #ccc;
            border-radius: 4px;
          }
        }

        .labels, .archive {
          h3 {
            display: flex;
            flex-direction: column;
            margin-left: 106px;
            margin-bottom: 20px;
          }
        }

        .follow-buttons {
          display: flex;
          flex-direction: column;
          align-items: center;
          gap: 10px;
          margin-top: 10px;
          margin-bottom: 45px;
        
          .follow1, .follow2 {
            display: flex;
            justify-content: center;
            align-items: center;
            padding: 10px;
            width: 50%;
            height: 30px;
            border: none;
            border-radius: 5px;
            cursor: pointer;
            color: #fff;
            font-size: 12px;
            font-weight: bold;
            text-align: center;
            transition: background-color 0.3s, transform 0.2s;
        
            &:hover {
              transform: scale(1.05);
            }
          }
        
          .follow1 {
            background: #007bff;
        
            &:hover {
              background: #0056b3;
            }
          }
        
          .follow2 {
            background: black;
        
            &:hover {
              background: #333;
            }
          }
        }                

        .learn-more {
          color: gray;
          text-align: center;
          padding: 172px;
          text-decoration: none;
        }

        .help-center {
          display: block;
          text-align: center;
          padding: 10px;
          color: darkblue;
          text-decoration: none;
        }
      }

      .hero {
        background: #f0f0f0;
        text-align: center;
        padding: 50px 0;
        font-weight: bold;
      }
      
      .hero h1 {
        font-size: 1.5em;
      }

      .blog-list {
        width: 70%;
        padding-left: 20px;

        .blog-item {
          display: flex;
          gap: 20px;
          margin-bottom: 20px;
          background-color: #fff;
          padding: 10px;

          img {
            width: 377px;
            height: 212px;
            object-fit: cover;
          }

          .blog-info {
            flex: 1;

            p {
              margin: 5px 0;
              color: gray;
            }

            h3 {
              margin: 10px 0;
              color: #333;
            }

            a {
              color: black;
              text-decoration: none;
            }
          }
        }
      }
    }
  }
}