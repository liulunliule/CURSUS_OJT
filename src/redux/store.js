import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
