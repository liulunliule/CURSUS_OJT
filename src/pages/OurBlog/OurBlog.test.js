import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from '../../redux/store';
import OurBlog from '.';
import '@testing-library/jest-dom/extend-expect';

const renderComponent = () =>
  render(
    <Provider store={store}>
      <MemoryRouter>
        <OurBlog />
      </MemoryRouter>
    </Provider>
  );

describe('OurBlog component', () => {
  test('renders OurBlog component correctly', () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  test('displays the correct hero section title', () => {
    renderComponent();
    const heroTitle = screen.getByText('Insights, ideas, and stories');
    expect(heroTitle).toBeInTheDocument();
  });

  test('renders loading state correctly', () => {
    renderComponent();
    jest.mock('./data', () => ({
      useFetchBlogs: () => ({
        blogs: [],
        status: 'loading',
        error: null,
      }),
    }));
  });

  test('renders error state correctly', () => {
    renderComponent();
    jest.mock('./data', () => ({
      useFetchBlogs: () => ({
        blogs: [],
        status: 'failed',
        error: 'Error fetching blogs',
      }),
    }));
  });

  test('displays blogs when data is fetched successfully', () => {
    renderComponent();
    jest.mock('./data', () => ({
      useFetchBlogs: () => ({
        blogs: [
          {
            Blog_ID: 1,
            Blog_Image: 'image1.jpg',
            Views: 100,
            Create_At: '2023-08-01',
            Blog_Title: 'First Blog',
            Intro: 'Introduction to the first blog',
          },
          {
            Blog_ID: 2,
            Blog_Image: 'image2.jpg',
            Views: 200,
            Create_At: '2023-08-02',
            Blog_Title: 'Second Blog',
            Intro: 'Introduction to the second blog',
          },
        ],
        status: 'succeeded',
        error: null,
      }),
    }));
  });

  test('search input renders correctly', () => {
    renderComponent();
    const searchInput = screen.getByPlaceholderText('Search');
    expect(searchInput).toBeInTheDocument();
  });

  test('labels and archive sections render correctly', () => {
    renderComponent();
    const labelsSection = screen.getByText('Labels');
    const archiveSection = screen.getByText('Archive');
    expect(labelsSection).toBeInTheDocument();
    expect(archiveSection).toBeInTheDocument();
  });

  test('follow buttons render correctly', () => {
    renderComponent();
    const followFacebookButton = screen.getByText('Follow our Facebook page');
    const followXButton = screen.getByText('Follow our X page');
    expect(followFacebookButton).toBeInTheDocument();
    expect(followXButton).toBeInTheDocument();
  });

  test('learn more and help center links render correctly', () => {
    renderComponent();
    const learnMoreLink = screen.getByText('Learn More');
    const helpCenterLink = screen.getByText('Cursus Help Center');
    expect(learnMoreLink).toBeInTheDocument();
    expect(helpCenterLink).toBeInTheDocument();
  });
});