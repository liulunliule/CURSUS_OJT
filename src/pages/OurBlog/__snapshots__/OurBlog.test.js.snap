// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OurBlog renders OurBlog correctly 1`] = `
<div>
  <header
    class="custom-header"
    style="background-color: #f8f8f8;"
  >
    <nav
      class="custom-nav"
    >
      <ul
        class="custom-ul"
      >
        <li
          class="custom-li"
        >
          About
        </li>
        <li
          class="custom-li selected"
        >
          Blog
        </li>
        <li
          class="custom-li"
        >
          Company
        </li>
        <li
          class="custom-li"
        >
          Careers
        </li>
        <li
          class="custom-li"
        >
          Press
        </li>
      </ul>
    </nav>
  </header>
  <section
    class="hero"
    style="background-color: #f8f8f8;"
  >
    <h1>
      Insights, ideas, and stories
    </h1>
  </section>
  <section
    class="content"
  >
    <div
      class="content-main"
    >
      <div
        class="sidebar"
      >
        <div
          class="left"
        >
          <div
            class="search"
          >
            <input
              type="text"
              placeholder="Search"
            />
          </div>
          <div
            class="labels"
          >
            <h3>
              Labels
            </h3>
          </div>
          <div
            class="archive"
          >
            <h3>
              Archive
            </h3>
          </div>
          <div
            class="follow-buttons"
          >
            <button
              class="follow1"
            >
              Follow our Facebook page
            </button>
            <button
              class="follow2"
            >
              Follow our X page
            </button>
          </div>
          <a
            href="#"
            class="learn-more"
          >
            Learn More
          </a>
          <a
            href="#"
            class="help-center"
          >
            Cursus Help Center
          </a>
        </div>
      </div>
      <div
        class="blog-list"
      >
        <div
          class="blog-item"
        >
          <img
            src="/path/to/image1.jpg"
            alt="Blog 1"
          />
          <div
            class="blog-info"
          >
            <p>
              123 • 2024-01-01
            </p>
            <h3>
              First Blog
            </h3>
            <p>
              This is the first blog.
            </p>
            <a
              href="/secondLayout/blog-detail"
            >
              Read More
            </a>
          </div>
        </div>
        <div
          class="blog-item"
        >
          <img
            src="/path/to/image2.jpg"
            alt="Blog 2"
          />
          <div
            class="blog-info"
          >
            <p>
              456 • 2024-01-02
            </p>
            <h3>
              Second Blog
            </h3>
            <p>
              This is the second blog.
            </p>
            <a
              href="/secondLayout/blog-detail"
            >
              Read More
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
